embedded-tests/src/test/java/org/apache/druid/testing/embedded/msq/EmbeddedMSQRealtimeQueryTest.java

```diff
@@ -0,0 +1,383 @@
+    final long selectedCount = Long.parseLong(msqApis.runDartSql(sql));
+
+    // By default Dart includes realtime data.
+    Assertions.assertEquals(totalRows, selectedCount);
+  }
+
+  @Test
+  @Timeout(60)
+  @Disabled // Test does not currently pass, see https://github.com/apache/druid/issues/18199
+  public void test_selectCount_dart_noRealtime()
+  {
+    final String sql = StringUtils.format(
+        "SET includeSegmentSource = 'NONE';\n"
+        + "SELECT COUNT(*) FROM \"%s\"",
+        dataSource
+    );
+
+    final long selectedCount = Long.parseLong(msqApis.runDartSql(sql));
+    Assertions.assertEquals(0, selectedCount);
+  }
+
+  @Test
+  @Timeout(60)
+  @Disabled // Test does not currently pass, see https://github.com/apache/druid/issues/18198
+  public void test_selectJoin_dart()
+  {
+    final long selectedCount = Long.parseLong(
+        msqApis.runDartSql(
+            "SELECT COUNT(*) FROM \"%s\"\n"
+            + "WHERE countryName IN (\n"
+            + "  SELECT countryName\n"
+            + "  FROM \"%s\"\n"
+            + "  WHERE countryName IS NOT NULL\n"
+            + "  GROUP BY 1\n"
+            + "  ORDER BY COUNT(*) DESC\n"
+            + "  LIMIT 1\n"
+            + ")",
+            dataSource,
+            dataSource
+        )
+    );
+
+    Assertions.assertEquals(528, selectedCount);
+  }
+
+  @Test
+  @Timeout(60)
+  @Disabled // Test does not currently pass, see https://github.com/apache/druid/issues/18198
+  public void test_selectJoin_task_withRealtime()
+  {
+    final String sql = StringUtils.format(
+        "SET includeSegmentSource = 'REALTIME';\n"
+        + "SELECT COUNT(*) FROM \"%s\"\n"
+        + "WHERE countryName IN (\n"
+        + "  SELECT countryName\n"
+        + "  FROM \"%s\"\n"
+        + "  WHERE countryName IS NOT NULL\n"
+        + "  GROUP BY 1\n"
+        + "  ORDER BY COUNT(*) DESC\n"
+        + "  LIMIT 1\n"
+        + ")",
+        dataSource,
+        dataSource
+    );
+
+    final MSQTaskReportPayload payload = msqApis.runTaskSql(sql);
+
+    BaseCalciteQueryTest.assertResultsEquals(
+        sql,
+        Collections.singletonList(new Object[]{528}),
+        payload.getResults().getResults()
+    );
+  }
+
+  private KafkaSupervisorSpec createKafkaSupervisor()
+  {
+    final Period startDelay = Period.millis(10);
+    final Period supervisorRunPeriod = Period.millis(500);
+    final boolean useEarliestOffset = true;
+
+    return new KafkaSupervisorSpec(
+        dataSource,
+        null,
+        DataSchema.builder()
+                  .withDataSource(dataSource)
+                  .withTimestamp(new TimestampSpec("__time", "auto", null))
+                  .withGranularity(new UniformGranularitySpec(Granularities.DAY, null, null))
+                  .withDimensions(DimensionsSpec.builder().useSchemaDiscovery(true).build())
+                  .build(),
+        null,
+        new KafkaSupervisorIOConfig(
```

Comment:
Should we add a builder or static utility method for creating a `KafkaSupervisor` and `KafkaSupervisorIOConfig` etc since several tests will need to use them?

Reply:
Fair enough, will keep it in mind when we start migrating the tests.